otherlibs/systhreads/thread.ml

```diff
@@ -92,3 +92,6 @@ let wait_pid p = Unix.waitpid [] p
 
 let sigmask = Unix.sigprocmask
 let wait_signal = Unix.sigwait
+
+external set_current_thread_name : string -> unit =
```

Comment:
Very nittily, but if the function is `set_current_thread_name` can the primitive be `caml_set_current_thread_name`, please?

Reply:
While we're at I find the `current_thread` part a tad verbose. We already have `Thread.self`, I think I prefer the suggested `set_self_name`. Any opinion?

Reply:
My non-native speaker brain feels like `set_current_thread_name` is more idiomatic/clear than `set_self_thread_name`.

Self is a notion of personal identity or belonging.

The call is made from the perspective of a programmer so, when reading the code, it's more logical to see "Set the current thread name" than "set self thread name" because, when reading the latter as the programmer, this means "set the name of the tread that belong to me".